index.js

```diff
@@ -1,68 +1,65 @@
+import { monsters } from './monsters.js';
 
-import {monsters} from './monsters.js';
-console.log(monsters);
-for(let monsterDiv of monsters){
-    showMonster(monsterDiv);
+const monsterContainer = document.querySelector('.monsters');
+const monsterFilter = document.querySelector('#search-monster');
 
+function populateMonsters(content) {
+  // Clear out existing content
+  monsterContainer.innerHTML = '';
+  // Usde showMonster to generate an Array of DOMNodes
+  let createMonsters = content.map((el) => showMonster(el));
+  // Append the DOMNodes in one DOM transaction
+  monsterContainer.append(...createMonsters);
 }
```

Comment:
Changing the entire approach here from Creating all DOM nodes and then flipping their displays, we are filtering the data at source > generating a fresh set of DOM nodes (monster) and updating the DOM in one shot. This is more efficient. DOM manipulation can be very expensive (slow)